reframe/core/pipeline.py

```diff
@@ -1220,6 +1220,7 @@ def compile(self):
             try:
                 self._build_job.prepare(
                     build_commands, environs,
+                    self._current_partition.preload_cmds,
```

Comment:
I would rename the commands everywhere to

```suggestion
                    self._current_partition.prepare_cmds,
```